avro-converter/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -278,10 +281,133 @@ public void testFromConnectComplex() {
   }
   
   @Test
-  public void testFromConnectOptionalWithDefaultNull() {
+  public void testFromConnectComplexWithDefaults() {
     Schema schema = SchemaBuilder.struct()
-        .field("optionalBool", SchemaBuilder.bool().optional().defaultValue(null).build())
+                                 .field("int8", SchemaBuilder.int8().defaultValue((byte) 2).doc("int8 field").build())
```

Comment:
I tend to like parameterized unit tests (such as in https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/test/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialectTypeTest.java) 

Not sure how many parameters this would need and how much it'd compress the code. Just FYI